syft/messaging/message.py

```diff
@@ -1014,8 +1014,10 @@ class CryptenInitPlan(Message):
     allows the exchange of information such as the ip and port of the master party to connect to,
     as well as the rank of the party to run and the number of parties involved."""
 
-    def __init__(self, crypten_context):
+    def __init__(self, crypten_context, model=None):
+        # crypten_context = (rank_to_worker_ids, world_size, master_addr, master_port)
         self.crypten_context = crypten_context
+        self.model = model
 
     def __str__(self):
         """Return a human readable version of this message"""
```

Comment:
CryptenInitPlan and CryptenInitJail converged to the same functionality at this point, right? Maybe we should rethink factorizing them into a single class?

Reply:
I think it is a good idea to keep them separated because we still have the ```jail_runner``` attribute that is only specific to Jails. I think keeping them separate it is better to decouple some of the logic for plans/jails (even if they look very similar)

Reply:
Absolutely, I also think they should be separate, but if they have pretty similar functionality then we can use inheritance maybe?

Reply:
Hmmm...We can do that. I opened a ticket for [it](https://github.com/OpenMined/PySyft/issues/3808)